python/gradbench/gradbench/comparison.py

```diff
@@ -26,6 +26,12 @@ def compare_json_objects(expected, actual, tolerance=1e-4, path=""):
     """
     mismatches = []
 
+    # Fudge the types a bit.
+    if type(expected) is int and type(actual) is float:
+        expected = float(expected)
+    if type(expected) is float and type(actual) is int:
+        actual = float(actual)
```

Comment:
The second seems fine: if `expected` is a `float` and `actual` is an `int`, it makes sense to coerce. But the first one doesn't seem as fine: if `expected` is an `int` and `actual` is a `float`, that seems like a mistake.

Reply:
The inputs to this function is currently always JSON objects, so the difference is pretty much just down to arbitrary implementation decisions in (de)serialisers.

Reply:
Not quite, right? Doesn't the `expected` come directly from some Python implementation running in-process, with no JSON (de)serialization?